modules/dataquery/php/view.class.inc

```diff
@@ -0,0 +1,311 @@
+                    'endkey'   => "[\"$category\", \"$field\", {}]",
+                )
+            );
+            $sessionResults = array_map(
+                function ($element) {
+                    return $element['value'];
+                },
+                $results
+            );
+            break;
+        case 'startsWith':
+            $results        = $cdb->queryView(
+                'DQG-2.0',
+                'search',
+                array(
+                    'reduce'   => 'false',
+                    'startkey' => "[\"$category\", \"$field\", \"$value\"]",
+                    'endkey'   => "[\"$category\", \"$field\", \"$value"
+                        // PHP doesn't have any /u9999 unicode escaping, so
+                        //  we use mb_convert_encoding to embed a high
+                        //  unicode character
+                        . mb_convert_encoding(
+                            '&#x9999;',
+                            'UTF-8',
+                            'HTML-ENTITIES'
+                        )
+                        . "\"]",
+                )
+            );
+            $sessionResults = array_map(
+                function ($element) {
+                    return $element['value'];
+                },
+                $results
+            );
+            break;
+        case 'contains':
+            $results        = $cdb->queryView(
+                'DQG-2.0',
+                'search',
+                array(
+                    'reduce'   => 'false',
+                    'startkey' => "[\"$category\", \"$field\"]",
+                    'endkey'   => "[\"$category\", \"$field\", {} ]",
+                )
+            );
+            $sessionResults = array_filter(
+                $results,
+                function ($element) use ($value) {
+                    /* Element is of the form:
+                    Array
+                        (
+                            [id] => 929342e78fd78dfe38e2732d7207b82e
+                            [key] => Array
+                            (
+                                [0] => adi_r_proband
+                                [1] => Administration
+                                [2] => All
+                            )
+                                [value] => Array
+                            (
+                                [0] => STL0138
+                                [1] => V06
+                            )
+                            );
+                     */
+                    $fieldVal = $element['key'][2];
+                    return strpos($fieldVal, $value) !== false;
+                }
+            );
+            $sessionResults = array_values(
+                array_map(
+                    function ($element) {
+                        return $element['value'];
+                    },
+                    $sessionResults
+                )
+            );
+            break;
+        default:
```

Comment:
In case of default, BadRequest with 'invalid operator' would be a good response.